lucid/overview.py

```diff
@@ -354,3 +361,74 @@ def add_from_dict(self, items):
         for location, dev_list in items.items():
             stand, system = location.split("|")
             self.add_devices(dev_list, stand=stand, system=system)
+
+
+class QuickAccessToolbar(QtWidgets.QWidget):
+    """Tab Widget with tabs containing buttons defined via a yaml file"""
+    def __init__(self, parent=None):
+        super().__init__(parent=parent)
+        self._tools = None
+        self.setSizePolicy(QtWidgets.QSizePolicy.Minimum,
+                           QtWidgets.QSizePolicy.Minimum)
+        self._setup_ui()
+
+    def sizeHint(self):
+        return QtCore.QSize(100, 100)
+
+    @Property(str)
+    def toolsFile(self):
+        return self._tools_file
+
+    @toolsFile.setter
+    def toolsFile(self, file):
+        if not file:
+            return
+        if isinstance(file, (str, bytes, os.PathLike)):
+            with open(self._tools_file, 'r') as tf:
+                self._tools = yaml.full_load(tf)
+        else:
+            self._tools = yaml.full_load(file)
+        self._assemble_tabs()
+
+    def _setup_ui(self):
+        main_layout = QtWidgets.QVBoxLayout()
+        self.setLayout(main_layout)
+        self.tab = QtWidgets.QTabWidget()
+        main_layout.addWidget(self.tab)
+
+    def _assemble_tabs(self):
+        self.tab.clear()
+        try:
+            for tab_name, tab_items in self._tools.items():
+                page = QtWidgets.QWidget()
+                page.setLayout(SnakeLayout(4))
+                for button in tab_items:
+                    for button_text, button_config in button.items():
+                        button_widget = self._button_factory(button_text,
+                                                             button_config)
+                        page.layout().addWidget(button_widget)
+                self.tab.addTab(page, tab_name)
+        except (IOError, ValueError):
+            logger.error('Invalid file for QuickAccessToolbar widget. %s',
```

Comment:
Where do these exceptions get raised?
Why does it complain about the file instead of something more specific here (e.g., the tab that caused the issue)?

```suggestion
            logger.exception('Invalid file for QuickAccessToolbar widget. %s',
```

Reply:
The exceptions were left-over from past code... I will remove it.